techmatters/flex-plugins
# [Client Profiles] CHI-2367 Customizable options for blocking timeframes

Primary reviewer:

## Description
- This PR allows for time based blocks on identifiers
- Set up "Short Term Block (24 Hours)" and "Long Term Block (5 Years)" options. For each option, `flag`, `label`, and `durationInHours` are defined to be used in the frontend
- Custom block durations are sorted and mapped within the UI, allowing administrators to choose and apply them easily.

### Checklist

### Related Issues
Fixes # [2367](https://tech-matters.atlassian.net/browse/CHI-2367), [2368](https://tech-matters.atlassian.net/browse/CHI-2368)

### Verification steps
- Run this branch for backend -  https://github.com/techmatters/hrm/pull/536
- Build `hrm-form-definitions`
- Run Aselo dev locally

- Check options in status flags like below


![Screenshot 2024-01-19 at 1 31 46 PM](https://github.com/techmatters/flex-plugins/assets/102122005/2a30b00a-696c-4992-9d00-d64c3e62972d)
![Screenshot 2024-01-19 at 1 31 55 PM](https://github.com/techmatters/flex-plugins/assets/102122005/760bdd0c-d89d-47a6-9a22-f5931336ab79)
